crates/blockchain/blockchain.rs

```diff
@@ -665,6 +674,14 @@ impl Blockchain {
         self.mempool.remove_transaction(hash)
     }
 
+    /// Remove all transactions in the executed block from the pool (if we have them)
```

Comment:
Do we really need this? See this PR: https://github.com/lambdaclass/ethrex/pull/4029

It seems the place to remove mempool tx should be the FCU call